action/protocol/context.go

```diff
@@ -43,6 +43,8 @@ type RunActionsCtx struct {
 	IntrinsicGas uint64
 	// Nonce is the nonce of the action
 	Nonce uint64
+	// History indicates whether to save account/contract history or not
+	History bool
```

Comment:
it is not a good idea to pass "History" all around the packages. I would strongly suggest we reconsider the current implementation. It is not a feature that in rush.